File: app/src/main/java/com/davismiyashiro/expenses/view/splitter/SplitterView.java
package com.davismiyashiro.expenses.view.splitter;

import com.davismiyashiro.expenses.datatypes.Expense;
import com.davismiyashiro.expenses.datatypes.Participant;

import java.util.List;

interface SplitterView {

    interface View {
        void splitWithGroup();
        void setParticipantSelectedError();
    }

    interface UserActionsListener {
        void validateParticipantsSelected (int size);
        List<Participant> getParticipants (Expense expense);
        void saveSplits (List<Participant> payers, Expense expense);
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/splitter/SplitterView.kt
package com.davismiyashiro.expenses.view.splitter

import com.davismiyashiro.expenses.datatypes.Expense
import com.davismiyashiro.expenses.datatypes.Participant

interface SplitterView {

    interface View {
        fun splitWithGroup()
        fun setParticipantSelectedError()
    }

    interface UserActionsListener {
        fun validateParticipantsSelected(size: Int)
        fun getParticipants(expense: Expense): List<Participant>
        fun saveSplits(payers: List<Participant>, expense: Expense)
    }
}
